Mathlib/RingTheory/LocalRing/ResidueField/Basic.lean

```diff
@@ -0,0 +1,199 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Chris Hughes, Mario Carneiro
+-/
+import Mathlib.RingTheory.LocalRing.ResidueField.Defs
+import Mathlib.RingTheory.LocalRing.RingHom.Basic
+import Mathlib.RingTheory.LocalRing.MaximalIdeal.Basic
+
+#align_import ring_theory.ideal.local_ring from "leanprover-community/mathlib"@"ec1c7d810034d4202b0dd239112d1792be9f6fdc"
+
+/-!
+
+# Residue Field of local rings
+
+We prove basic properties of the residue field of a local ring.
+
+-/
+
+
+variable {R S T : Type*}
+
+namespace LocalRing
+
+section
+
+variable (R) [CommRing R] [LocalRing R] [CommRing S] [LocalRing S] [CommRing T] [LocalRing T]
+
+-- Porting note: failed at `deriving` instances automatically
+instance ResidueFieldCommRing : CommRing (ResidueField R) :=
+  show CommRing (R ⧸ maximalIdeal R) from inferInstance
+
+instance ResidueFieldInhabited : Inhabited (ResidueField R) :=
+  show Inhabited (R ⧸ maximalIdeal R) from inferInstance
+
+noncomputable instance ResidueField.field : Field (ResidueField R) :=
+  Ideal.Quotient.field (maximalIdeal R)
+#align local_ring.residue_field.field LocalRing.ResidueField.field
+
+/-- The quotient map from a local ring to its residue field. -/
+def residue : R →+* ResidueField R :=
+  Ideal.Quotient.mk _
+#align local_ring.residue LocalRing.residue
```

Comment:
These are all defining data. If it doesn't increase imports, I think it would be good to have these in `Defs.lean`.

Reply:
I've moved all of them.  `ResidueField.field` needs `LocalRing.MaximalIdeal.Basic` instead of `LocalRing.MaximalIdeal.Defs`, but I think this is fine (I can move it back if we want to only import `LocalRing.MaximalIdeal.Defs`).